### PR TITLE
Makes Token API public.

### PR DESCRIPTION
**Pull request by achrafmam2**
Makes Token methods public
Makes convertToken(_ clang: CXToken) function public.
Make clang property public.

### CODE REVIEW

File: Sources/Clang/Token.swift [hunk 1]
```diff
@@ -114,7 +114,7 @@ public struct SourceLocation {
 
 /// Represents a half-open character range in the source code.
 public struct SourceRange {
-  let clang: CXSourceRange
```
Comment (CodaFi) [inline Sources/Clang/Token.swift]: We generally don’t expose the C API accessors to users if we can help it.  Is there a reason you need this?

Comment (achrafmam2) [reply Sources/Clang/Token.swift]: I saw that PunctuationToken for example that it exposed the clang property.  I opted for consistency in the file.

In the case of Token the clang property seems useful in order to use the convertToken(_ clang: CXToken) function.

Maybe I think it is better to use an asClang() method?

Comment (CodaFi) [reply Sources/Clang/Token.swift]: A lot of this file was auto-generated before we established the hazy guidelines we follow in LLVMSwift.  I think providing `asClang()` would be a better way, yes.

File: Sources/Clang/Token.swift [hunk 2]
```diff
@@ -61,7 +66,7 @@ public struct CommentToken: Token {
 }
 
 /// Converts a CXToken to a Token, returning `nil` if it was unsuccessful
-func convertToken(_ clang: CXToken) -> Token {
+public func convertToken(_ clang: CXToken) -> Token {
```
Comment (CodaFi) [inline Sources/Clang/Token.swift]: This one also shouldn't be public.  In theory you shouldn't ever see `CXToken` as a consumer of this framework.

Comment (achrafmam2) [reply Sources/Clang/Token.swift]: Done.

File: Sources/Clang/Token.swift (hunk 1 above)
Comment (achrafmam2) [reply Sources/Clang/Token.swift]: Thanks.

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**